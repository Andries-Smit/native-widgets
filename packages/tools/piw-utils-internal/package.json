{
  "name": "@mendix/piw-utils-internal",
  "version": "1.0.0",
  "description": "Tools and helper functions for PIW",
  "license": "Apache-2.0",
  "copyright": "Â© Mendix Technology BV 2022. All rights reserved.",
  "homepage": "https://github.com/mendix/native-widgets/tree/main/packages/tools/piw-utils-internal",
  "repository": {
    "type": "git",
    "url": "https://github.com/mendix/native-widgets/"
  },
  "private": true,
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "start": "tsc --watch",
    "prebuild": "rimraf ./dist",
    "build": "tsc",
    "lint": "eslint --config ../../../.eslintrc.js --ext .jsx,.js,.ts,.tsx src/",
    "prepare": "pnpm build",
    "test": "pluggable-widgets-tools test:unit:native",
    "postinstall": "pnpm build"
  },
  "devDependencies": {
    "@mendix/pluggable-widgets-tools": "*",
    "rimraf": "^5.0.10"
  },
  "files": [
    "components",
    "dist"
  ]
}
