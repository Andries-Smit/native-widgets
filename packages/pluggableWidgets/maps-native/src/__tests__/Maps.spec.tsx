/**
 * @jest-environment jsdom
 */
import { Maps, Props } from "../Maps";
import { mount } from "enzyme";
import { dynamicValue } from "@mendix/piw-utils-internal";

describe("", () => {
    let defaultProps: Props;

    beforeEach(() => {
        defaultProps = {
            name: "maps-test",
            style: [],
            markers: [],
            dynamicMarkers: [],
            fitToMarkers: true,
            defaultZoomLevel: "city",
            minZoomLevel: "city",
            maxZoomLevel: "streets",
            mapType: "standard",
            provider: "default",
            interactive: true,
            showsUserLocation: false
        };
    });

    it("renders", async () => {
        defaultProps.markers = [
            {
                locationType: "address",
                address: dynamicValue<string>("Tokyo", false),
                iconSize: 1,
                iconColor: "red"
            },
            {
                locationType: "address",
                address: dynamicValue<string>("Amsterdam", false),
                iconSize: 1,
                iconColor: "red"
            }
        ];
        const wrapper = mount(<Maps {...defaultProps} />);
        // wrapper.setState({ status: "mapReady" });
        await new Promise(resolve => {
            setTimeout(resolve, 5000);
        });

        wrapper.update();
        console.log(wrapper.state.status);
        // wrapper.find("MarkerView")
        expect(wrapper).toMatchSnapshot();
    });
});
