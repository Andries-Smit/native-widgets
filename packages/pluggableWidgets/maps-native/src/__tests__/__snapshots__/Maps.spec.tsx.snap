// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Maps
  defaultZoomLevel="city"
  dynamicMarkers={Array []}
  fitToMarkers={true}
  interactive={true}
  mapType="standard"
  markers={
    Array [
      Object {
        "address": Object {
          "status": "available",
          "value": "Tokyo",
        },
        "iconColor": "red",
        "iconSize": 1,
        "locationType": "address",
      },
      Object {
        "address": Object {
          "status": "available",
          "value": "Amsterdam",
        },
        "iconColor": "red",
        "iconSize": 1,
        "locationType": "address",
      },
    ]
  }
  maxZoomLevel="streets"
  minZoomLevel="city"
  name="maps-test"
  provider="default"
  showsUserLocation={false}
  style={Array []}
>
  <Component
    style={
      Object {
        "minHeight": 50,
      }
    }
    testID="maps-test"
  >
    <View
      style={
        Object {
          "minHeight": 50,
        }
      }
      testID="maps-test"
    >
      <MapView
        cacheEnabled={false}
        liteMode={false}
        mapPadding={
          Object {
            "bottom": 48,
            "left": 48,
            "right": 48,
            "top": 48,
          }
        }
        mapType="standard"
        maxZoomLevel={15}
        minZoomLevel={3}
        onMapReady={[Function]}
        onRegionChangeComplete={[Function]}
        pitchEnabled={false}
        provider={null}
        rotateEnabled={true}
        scrollEnabled={true}
        showsMyLocationButton={false}
        showsPointsOfInterest={false}
        showsTraffic={false}
        showsUserLocation={false}
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
        zoomEnabled={true}
      >
        <AIRMap
          cacheEnabled={false}
          handlePanDrag={false}
          initialRegion={null}
          liteMode={false}
          mapPadding={
            Object {
              "bottom": 48,
              "left": 48,
              "right": 48,
              "top": 48,
            }
          }
          mapType="standard"
          maxZoomLevel={15}
          minZoomLevel={3}
          onChange={[Function]}
          onLayout={[Function]}
          onMapReady={[Function]}
          onMarkerPress={[Function]}
          onRegionChangeComplete={[Function]}
          pitchEnabled={false}
          provider={null}
          region={null}
          rotateEnabled={true}
          scrollEnabled={true}
          showsMyLocationButton={false}
          showsPointsOfInterest={false}
          showsTraffic={false}
          showsUserLocation={false}
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
          zoomEnabled={true}
        >
          <AIRMap
            cacheEnabled={false}
            handlePanDrag={false}
            initialRegion={null}
            liteMode={false}
            mapPadding={
              Object {
                "bottom": 48,
                "left": 48,
                "right": 48,
                "top": 48,
              }
            }
            mapType="standard"
            maxZoomLevel={15}
            minZoomLevel={3}
            onChange={[Function]}
            onLayout={[Function]}
            onMapReady={[Function]}
            onMarkerPress={[Function]}
            onRegionChangeComplete={[Function]}
            pitchEnabled={false}
            provider={null}
            region={null}
            rotateEnabled={true}
            scrollEnabled={true}
            showsMyLocationButton={false}
            showsPointsOfInterest={false}
            showsTraffic={false}
            showsUserLocation={false}
            style={
              Object {
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
            zoomEnabled={true}
          />
        </AIRMap>
      </MapView>
    </View>
  </Component>
</Maps>
`;
